utils/catalog_production/add_glsn_properties.py

```diff
@@ -0,0 +1,286 @@
+import numpy as np
+import pandas as pd
+import os
+import sys
+import GCRCatalogs
+import pandas as pd
+from GCR import GCRQuery
+from lsst.sims.photUtils import Sed, Bandpass, BandpassDict
+sys.path.append('/global/homes/b/brycek/DC2/sims_GCRCatSimInterface/workspace/sed_cache/')
+from SedFitter import sed_from_galacticus_mags
+
+
+def load_limited_cat(dc2_df_system, catalog):
+
+    z_cat_min = np.power(10, np.log10(np.min(dc2_df_system['zl'])) - .1)
+    z_cat_max = np.power(10, np.log10(np.max(dc2_df_system['zl'])) + .1)
+
+    i_gal_dim = np.max(-2.5*np.log10(np.power(10, -0.4*dc2_df_system['lensgal_mi'])*.9))
+    i_gal_bright = np.min(-2.5*np.log10(np.power(10, -0.4*dc2_df_system['lensgal_mi'])*1.1))
+
+    print(z_cat_min, z_cat_max, i_gal_dim, i_gal_bright)
+
+    data = catalog.get_quantities(['galaxy_id', 'redshift_true', 'ellipticity_true', 'mag_true_i_lsst',
+                                   'size_minor_bulge_true', 'size_bulge_true'],
+                                  filters=['mag_true_i_lsst > %f' % i_gal_bright,
+                                           'mag_true_i_lsst < %f' % i_gal_dim,
+                                           'redshift_true > %f' % z_cat_min,
+                                           'redshift_true < %f' % z_cat_max,
+                                           'stellar_mass_bulge/stellar_mass > 0.99'])
+
+    data['glsne_ellipticity'] = (1-(data['size_minor_bulge_true']/data['size_bulge_true']))
+
+
+    data_df = pd.DataFrame(data)
+    data_df.to_csv('sn_matching_checkpoint_1.csv', index=False)
+
+
+def match_up_glsn(dc2_df_system, data_df):
+
+
+    gcr_glsn_match = []
+    err = 0
+    np.random.seed(10)
```

Comment:
Alternatively: I don't know if you want to make an `rng` kwarg in all of these methods so that you can pass a single random number generator around.  That would make sure that every random number draw is random wrt previous draws in other methods.  This comment would apply to the methods in the other scripts, as well.